### PR TITLE
Add Brave gulp config

### PR DESCRIPTION
**Pull request by bbondy**
It's easiest to review this commit by commit.

The change to ./gulpfile.js will be upstreamed so that we have no changes to that file in the end.

### CODE REVIEW

File: brave/gulpfile.js/common-gulp.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const gulp = require('gulp')
```
Comment (bbondy) [inline brave/gulpfile.js/common-gulp.js]: This file just copies some of the ./gulpfile.js files. I hope to eventually upstream a change to split them out into another file upstream and then we can just use that.

File: README.md [hunk 2]
```diff
@@ -43,13 +43,14 @@ You can run the linter by itself with `gulp lint`.
 
 ```bash
 npm install
-npm start
+npm dev:brave
```
Comment (ryanml) [inline README.md]: Should be `npm run dev:brave`?

File: brave/gulpfile.js/brave-locales.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const gulp = require('gulp')
+const jsoneditor = require('gulp-json-editor')
+
+const createBraveLocalesTask = () =>
+  gulp.task('locales:brave', function () {
+    return gulp.src('./dist/brave/_locales/**/*')
+    .pipe(jsoneditor(function (json) {
+      Object.keys(json).forEach((stringName) => {
+        if (json[stringName]) {
```
Comment (ryanml) [inline brave/gulpfile.js/brave-locales.js]: Just curious, what makes this check necessary? Are some props set to `undefined` or `false`?

Comment (bbondy) [reply brave/gulpfile.js/brave-locales.js]: it might not be necessary, I was probably just being overly robust, I'll remove it.

Comment (bbondy) [reply brave/gulpfile.js/brave-locales.js]: fixed

**Pull request merged**